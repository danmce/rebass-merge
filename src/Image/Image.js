import React from 'react'
import PropTypes from 'prop-types'
import { Image as RebassImage } from 'rebass'

const Image = ({ src, ...props }) => <RebassImage src={src} {...props} />

Image.propTypes = {
  src: PropTypes.string,
  variant: PropTypes.oneOf(['avatar'])
}

export default Image
