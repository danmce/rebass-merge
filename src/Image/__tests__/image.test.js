import React from 'react'
import * as enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import Image from '../Image'

test('renders Image correctly', () => {
  const image = (
    <Image
      src='https://fillmurray.com/200/300'
      sx={{
        width: ['100%', '50%'],
        borderRadius: 8
      }}
    />
  )

  expect(toJson(enzyme.mount(image))).toMatchSnapshot('enzyme.mount')
})
