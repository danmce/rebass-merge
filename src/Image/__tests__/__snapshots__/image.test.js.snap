// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Image correctly: enzyme.mount 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  width: 100%;
  border-radius: 8px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    width: 50%;
  }
}

<Image
  src="https://fillmurray.com/200/300"
  sx={
    Object {
      "borderRadius": 8,
      "width": Array [
        "100%",
        "50%",
      ],
    }
  }
>
  <ForwardRef
    src="https://fillmurray.com/200/300"
    sx={
      Object {
        "borderRadius": 8,
        "width": Array [
          "100%",
          "50%",
        ],
      }
    }
  >
    <Styled(div)
      __css={
        Object {
          "height": "auto",
          "maxWidth": "100%",
        }
      }
      as="img"
      src="https://fillmurray.com/200/300"
      sx={
        Object {
          "borderRadius": 8,
          "width": Array [
            "100%",
            "50%",
          ],
        }
      }
    >
      <img
        className="emotion-0"
        src="https://fillmurray.com/200/300"
      />
    </Styled(div)>
  </ForwardRef>
</Image>
`;
